packages/config/src/index.ts

```diff
@@ -178,6 +178,21 @@ class TruffleConfig {
   public static detect(options: any = {}, filename?: string): TruffleConfig {
     let configFile;
     const configPath = options.config;
+    const url = options.url;
+
+    if (url) {
```

Comment:
I don't understand the implications of this being a general feature option, but it absolutely makes sense to be scoped to the debugger CLI as that's the use case specified by #4295, which is an absolutely brilliant one.

Reply:
Makes sense. I agree that adding it to the debugger CLI might make more sense.

What about the possible issues with default configs you just mentioned? I am wondering what can cause problem and if we should combine `--url` always with `--compile-none` for example.
Any thoughts here?

Reply:
I mean I'm pretty the solution here, is, as mentioned, to merge the new `networks` into the existing config, *not* into a default one.  (Although as part of the idea is that this should work without a config file present, I guess it would use a default one if none exists?)  As I also mentioned, this probably doesn't belong in `Config.detect`, but rather in `core/lib/commands/debug`, around the call to `Config.detect`.  (I guess you'd have to catch errors from the `detect` to handle the no-config case?)

> I am wondering what can cause problem

I mean, Solidity versions are one example; I don't know of others *offhand*, but like -- it's far safer to make specific modifications to the existing config, rather than creating a new one that ignores the existing one entirely.

> if we should combine --url always with --compile-none for example.

No, absolutely not.  The `--compile-none` option is explicitly marked as dangerous; it's not something you would ever want turned on by accident just as a result of using `--url`.

Reply:
I did quite some changes. It seems to look better now.

1. `url` param is scoped for the `debug` command only
2. `compileNone` is set to `true` by default
3. user can override `network` inline if they want
4. added tests to debug command